clients/drcachesim/analyzer.h

```diff
@@ -114,6 +116,41 @@ template <typename RecordType, typename ReaderType> class analyzer_tmpl_t {
 protected:
     typedef scheduler_tmpl_t<RecordType, ReaderType> sched_type_t;
 
+    // Tool data for one shard.
+    struct analyzer_tool_shard_data_t {
+        analyzer_tool_shard_data_t()
+            : shard_data(nullptr)
+        {
+        }
+        analyzer_tool_shard_data_t(analyzer_tool_shard_data_t &&src)
+        {
+            shard_data = src.shard_data;
+            interval_snapshot_data = std::move(interval_snapshot_data);
+        }
+
+        void *shard_data;
+        // This is a queue as merge_shard_interval_results processes the intervals in a
+        // FIFO manner. Using a queue also makes code a bit simpler.
+        std::queue<typename analysis_tool_tmpl_t<RecordType>::interval_state_snapshot_t *>
+            interval_snapshot_data;
+    };
+
+    // Data for one trace shard.
+    struct analyzer_shard_data_t {
+        analyzer_shard_data_t()
+            : cur_interval_index(0)
+        {
+        }
+        analyzer_shard_data_t(analyzer_shard_data_t &&src)
+        {
+            cur_interval_index = src.cur_interval_index;
+            tool_data = std::move(tool_data);
```

Comment:
Is this line intended to be `tool_data = std::move(src.tool_data)`?

Note: I'm compiling DynamoRIO under LicheePi4A (RISC-V SBC), the compiler is complaining about this.

Reply:
Sent #6058 to fix this.